daemon/container_linux.go

```diff
@@ -306,6 +307,24 @@ func populateCommand(c *Container, env []string) error {
 	return nil
 }
 
+func mergeDevices(defaultDevices, userSpecifiedDevices []*configs.Device) []*configs.Device {
+	if len(userSpecifiedDevices) == 0 {
+		return defaultDevices
+	}
+
+	// Only add those default devices, which are not specified by the user.
+	// Otherwise, the user specified devices end up without permissions.
+	var devs []*configs.Device
+	for _, defaultDevice := range defaultDevices {
+		for _, userSpecified := range userSpecifiedDevices {
+			if defaultDevice.Path != userSpecified.Path {
```

Comment:
It adds all `defaultDevices` except those which are in `userSpecifiedDevices`. In the last line it adds the `userSpecifiedDevices`.

It merges `defaultDevices` and `userSpecifiedDevices`, but prefers a device in `userSpecifiedDevices` over a device in `defaultDevices`.
Say, there is `{"/dev/fuse", "/dev/null"}`in `defaultDevices` and `{"/dev/fuse"}`in `userSpecifiedDevices`, you get `{"/dev/null", "/dev/fuse"}` where `"/dev/fuse"` is taken from `userSpecifiedDevices`.

Does this make sense?